### PR TITLE
Reader search screen doesn't give any feedback to the user when there's no internet connection.

### LINKED ISSUE #16997: Reader search screen doesn't give any feedback to the user when there's no internet connection.
### Expected behavior
While searching for Posts and Sites in Reader, the user would be let know if there's no internet connection.

### Actual behavior
If the internet connection is lost while searching for Posts and Sites in Reader, there's no feedback to the user.

|  |
| :-: |
| <video width="300" src="https://user-images.githubusercontent.com/42008628/182728021-d1b93ecf-1caf-4333-84d5-e44fc8c6bb51.mp4"> | 

## Steps to reproduce the behavior
1. Airplane mode ON
2. Go to `Reader` -> `Search`
3. Search for any keyword
4. ❌ Notice that there's no feedback about connectivity.

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #16997

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Before:

https://user-images.githubusercontent.com/16153555/183618673-c19e9b1d-5e1f-489d-8ad6-7fdcabae0645.mp4

After:
https://www.loom.com/share/899fd240694e4f9bbb39dc50faec036d

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 1]
```diff
@@ -1315,6 +1325,11 @@ private void submitSearchQuery(@NonNull String query) {
         hideSearchMessage();
         hideSearchSuggestions();
 
+        if (!NetworkUtils.isNetworkAvailable(getContext())) {
+            resetPostAdapter(mPostListType);
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: Why did you add `resetPostAdapter(mPostListType)` here? I removed that line and couldn't find any difference on testing.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 2]
```diff
@@ -1315,6 +1325,11 @@ private void submitSearchQuery(@NonNull String query) {
...
+            showEmptyView();
+        }
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java:1330]: Do we need to continue `submitSearchQuery` function after `showEmptyView()`? Maybe `return` from here could be better. Can you test it?

**Review by irfano — CHANGES_REQUESTED** [review]
Thank you for working on this issue @gitstart! Please review my comments on the code.
I watched your after video, but there is no illustration above "No network available" message in your video. Is the video from the latest version of the PR? When I tested the PR, I saw the illustration as expected.

**Comment by gitstart** [comment]
Thank you @irfano for reviewing the PR, 
Added updated loom video and pushed requested changes. 
Please have a look. 🙌

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java (hunk 2 above)
Comment (gitstart) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java:1330]: If we add a return, we get the wrong screen shown in the image.

image

**Review by irfano — APPROVED** [review]
I couldn't understand the latest commits. I guess they have been pushed accidentally. But the diff is okay.
LGTM! 👍🏻

**Pull request merged**